src/plugins/es_ui_shared/static/forms/helpers/field_validators/is_integer.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { ValidationFunc } from '../../hook_form_lib';
+import { ERROR_CODE } from './types';
+
+export const isInteger =
+  ({ message }: { message: string }) =>
+  (...args: Parameters<ValidationFunc>): ReturnType<ValidationFunc<any, ERROR_CODE>> => {
+    const [{ value }] = args;
+
+    if (value === '') {
+      return;
+    }
+
+    const numValue = typeof value === 'string' ? Number(value) : value;
+
+    return Number.isInteger(numValue) ? undefined : { message, code: 'ERR_NOT_INT_NUMBER' };
```

Comment:
I am thinking of a case where this might not work properly - if `value` is `true` or `false`, Number(value) would evaluate to a number 1 or 0, which is an integer.

From what I understand, we only return `undefined` in these three cases:
- If `value` is undefined or an empty string
- If `value` is of type number and it is an integer
- If `value` is of type string and represents an integer
Otherwise returning the error message.

Following the above rules, I think the logic can be simplified to something like this:

```ts
  if (
    !value || 
    (typeof value === 'number' && Number.isInteger(value)) || 
    (typeof value === 'string' && Number.isInteger(Number(value)))
  ) {
    return undefined;
  }

  return { message, code: 'ERR_NOT_INT_NUMBER' };
```

WDYT?

Reply:
Thank for your comments, Elena!

I don't think it would be a problem in the case of a boolean. If value is a boolean it won't do `Number(value)` because it only do so for` typeof string`. If type is a boolean the evaluation will be `Number.isInteger(false) ` and then will return `{ message, code: 'ERR_NOT_INT_NUMBER' }` (I now realize that the name of the const `numValue ` is not good because it's not always a number, like this boolean case) 

In any case, probably your option is more precise so I can change it. And I'll add the boolean cases in the UTs too, I think they are a good case to be tested.